planning/obstacle_cruise_planner/src/polygon_utils.cpp

```diff
@@ -134,6 +134,25 @@ Polygon2d createOneStepPolygon(
   return hull_polygon;
 }
 
+std::optional<double> calcCollisionDistanceForStopObstacle(
+  const std::vector<TrajectoryPoint> & traj_points, const std::vector<Polygon2d> & traj_polys,
+  const Obstacle & obstacle)
+{
+  auto obstacle_polygon = tier4_autoware_utils::toPolygon2d(obstacle.pose, obstacle.shape);
+  for (size_t col_i = 0; col_i < traj_polys.size(); ++col_i) {
+    if (boost::geometry::intersects(traj_polys.at(col_i), obstacle_polygon)) {
+      try {
```

Comment:
Why do you use try here?
I think we can handle the corner case without using try, which is a better implementation